agent/config/config.go

```diff
@@ -257,6 +258,10 @@ type Config struct {
 	VerifyServerHostname *bool                    `json:"verify_server_hostname,omitempty" hcl:"verify_server_hostname" mapstructure:"verify_server_hostname"`
 	Watches              []map[string]interface{} `json:"watches,omitempty" hcl:"watches" mapstructure:"watches"`
 
+	RPC RPC `mapstructure:"rpc"`
+
+	ServiceHealthStreamingBackend *bool `json:"service_health_streaming_backend" hcl:"service_health_streaming_backend" mapstructure:"service_health_streaming_backend"`
```

Comment:
Maybe just `enable_streaming_backend`? Unless we want to specifically let people enable individual cachetypes once there are more, it seems like one setting for this would be ok.

Reply:
My thinking was that it would be nice to enable this by default before everything is ported to streaming. So as we add more streaming endpoints we could add new fields for each endpoint, while other endpoints were already default enabled.

I'm not sure how well that plan will work out, especially with the need to ensure that the Servers have the setting enabled first.

I think you're right that this flag is overly specific for where it is in the configuration structure. If we wanted more fine grained flags we should have a section for feature flags that are intended to be short lived and  that we can start ignoring as we default enable things.